src/Client/Client.php

```diff
@@ -79,6 +95,12 @@ public function handleEvent($event, $name = null)
             }
         }
 
+        if (null !== $this->toSendLimit && $this->getToSend()->count() >= $this->toSendLimit) {
+            $this->send();
+
+            return;
```

Comment:
With this case, i can't ask for an immediateSend if toSendLimit is set.
And the unit tests dont try this configuration

(okay, it's a nonsense use case, but it technically possible)